File: MigrationExample/entry/src/main/java/com/minkiapps/hos/migration/di/modules/AppModule.java
package com.minkiapps.hos.migration.di.modules;

import com.minkiapps.hos.migration.util.LoggerFacade;
import com.minkiapps.shared.util.Logger;
import dagger.Module;
import dagger.Provides;

@Module
public class AppModule {

    @Provides
    public static Logger providesLogger() {
        return new Logger(new LoggerFacade());
    }

}

File: AndroidExample/app/src/main/java/com/minkiapps/android/example/di/modules/AppModule.kt
package com.minkiapps.android.example.di.modules

import com.minkiapps.android.example.util.LoggerFacade
import com.minkiapps.shared.util.Logger
import dagger.Module
import dagger.Provides

@Module
class AppModule {

    @Provides
    fun providesLogger(): Logger {
        return Logger(LoggerFacade())
    }
}